### ISSUE TITLE
Compiler error concerning ITK_THREAD_RETURN_VALUE

### ISSUE THREAD

**Issue by jhlegarreta** [InsightSoftwareConsortium/LesionSizingToolkit#20]
There seems to be a compiler error involving the `ITK_THREAD_RETURN_VALUE` define:
```
Error C2065 'ITK_THREAD_RETURN_VALUE': undeclared identifier itkCannyEdgeDetectionImageFilter2  
\lesionsizingtoolkit\include\itkCannyEdgeDetectionRecursiveGaussianImageFilter.hxx 305
```

@dzenanz could you please cast some light on this?

Also missing to use the `ProgressTransformer` class instead of `ProgressReporter`.

**Comment by dzenanz** [comment]
LesionSizingToolkit has its own version of Canny filter. This filter has not been updated to use the new threading model. I don't know what the difference was (if any) when compared to normal Canny filter. Anyway, this filter will have to be rewritten to inherit from the original Canny, or updated in a similar way to original Canny. Canny filter underwent all the standard refactoring, plus [this](https://github.com/InsightSoftwareConsortium/ITK/commit/e086233e890f7dc00465185aa2dce99daeba058b) specific simplification.

**Comment by jhlegarreta** [comment]
You are right, the module would need some work to update it, adapt to the changes since 6 years now, avoid duplicates, and eventually get its way into the main toolkit. 

At this time, I'll just do the minimal changes to make it build against ITK v5. I'll add a *WIP* branch with CI, and once that is completed, we will be a in a better position to focus on further improvements.
